website/source/guides/load-balancing/haproxy.html.md

```diff
@@ -0,0 +1,272 @@
+          path     = "/"
+          interval = "2s"
+          timeout  = "2s"
+        }
+      }
+    }
+  }
+}
+```
+
+Note that this job deploys 3 instances of our demo web application which we will
+load balance with HAProxy in the next few steps.
+
+### Step 2: Deploy the Demo Web App
+
+We can now deploy our demo web application:
+
+```shell
+$ nomad run webapp.nomad 
+==> Monitoring evaluation "8f3af425"
+    Evaluation triggered by job "demo-webapp"
+    Evaluation within deployment: "dc4c1925"
+    Allocation "bf9f850f" created: node "d16a11fb", group "demo"
+    Allocation "25e0496a" created: node "b78e27be", group "demo"
+    Allocation "a97e7d39" created: node "01d3eb32", group "demo"
+    Evaluation status changed: "pending" -> "complete"
+==> Evaluation "8f3af425" finished with status "complete"
+```
+
+### Step 3: Create a Job for HAProxy
+
+Create a job for HAProxy and name it `haproxy.nomad`. This will be our load
+balancer that will balance requests to the deployed instances of our web
+application.
+
+```hcl
+job "haproxy" {
+  region      = "global"
+  datacenters = ["dc1"]
+  type        = "service"
+
+  group "haproxy" {
+    count = 1
+
+    task "haproxy" {
+      driver = "docker"
+
+      config {
+        image        = "haproxy:2.0"
+        network_mode = "host"
+
+        port_map {
+          http = 8080
+        }
+
+        volumes = [
+          "local/haproxy.cfg:/usr/local/etc/haproxy/haproxy.cfg",
+        ]
+      }
+
+      template {
+        data = <<EOF
+defaults
+   mode http
+
+frontend stats
+   bind *:1936
+   stats uri /
+   stats show-legends
+   no log
+
+frontend http_front
+   bind *:8080
+   default_backend http_back
+
+backend http_back
+    balance roundrobin
+    server-template mywebapp 10 _demo-webapp._tcp.service.consul resolvers consul resolve-opts allow-dup-ip resolve-prefer ipv4 check
+
+resolvers consul
+  nameserver consul 127.0.0.1:53
+  accepted_payload_size 8192
+  hold valid 5s
+EOF
+
+        destination = "local/haproxy.cfg"
+      }
+
+      service {
+        name = "haproxy"
+        port = "http"
+      }
+
+      resources {
+        cpu    = 200
+        memory = 128
+
+        network {
+          mbits = 10
+
+          port "http" {
+            static = 8080
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+Take note of the following key points from the HAProxy configuration we have
+defined:
+
+- The `backend http_back` stanza in the HAProxy config includes two main
+  settings: `balance type` and `server-template`. The `balance type` is round
+  robin and will load balance across the available service in order.
+- The `server-template` option allows Consul service registrations to configure
+  HAProxy's backend server pool. Because of this, you do not need to explicitly
+  add your backend servers' IP addresses. We have specified a server-template
+  named mywebapp. The template name is not tied to the service name which is
+  registered in Consul.
```

Comment:
I'd break the `balance type` and `server-template` discussions into dedicated bullets... or maybe even "demote" the first bullet to paragraph text, putting the two explanations into their own list.